common_drivers/platform_drivers/platform_drivers.c

```diff
@@ -196,6 +196,7 @@ int32_t spi_init(spi_desc **desc,
 		break;
 #endif
 	default:
+		free(dev);
```

Comment:
this is good;
but i think you may also need to add the` free()` at lines 210-211 & 214-215 below